### PR TITLE
Gridding rework

### PR DESCRIPTION
**Pull request by sjperkins**
- [x] Tests added / passed

  ```bash
  $ py.test -v -s africanus
  ```

  If the pycodestyle tests fail, the quickest way to correct
  this is to run `autopep8` and then `pycodestyle` to fix the
  remaining issues.

  ```
  $ pip install -U autopep8 pycodestyle
  $ autopep8 -r -i africanus
  $ pycodestyle africanus
  ```

- [x] Fully documented, including `HISTORY.rst` for all changes
      and one of the `docs/*-api.rst` files for new API

  To build the docs locally:

  ```
  pip install -r requirements.readthedocs.txt
  cd docs
  READTHEDOCS=True make html
  ```

### CODE REVIEW

File: africanus/gridding/simple/tests/test_simple_gridder.py [hunk 1]
```diff
@@ -78,7 +78,12 @@ def test_degridder_gridder():
 
 
 @pytest.mark.parametrize("plot", [False])
-def test_psf_subtraction(plot):
+@pytest.mark.parametrize("oversample", [
+    1,
+    pytest.param(2, marks=pytest.mark.xfail(reason="Unknown")),
+    pytest.param(3, marks=pytest.mark.xfail(reason="Unknown")),
+])
+def test_psf_subtraction(plot, oversample):
```
Comment (sjperkins) [inline africanus/gridding/simple/tests/test_simple_gridder.py]: @landmanbester This test succeeds with an oversampling of 1, but fails with 2 and 3.

File: africanus/filters/tests/test_oversampling.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+from africanus.gridding.simple.gridding import _ARCSEC2RAD
+
+
+@pytest.mark.parametrize("exact_u", [np.array([1.9])])
+@pytest.mark.parametrize("full_support, oversampling, beta", [[7, 7, 2.4]])
+@pytest.mark.parametrize("ref_wave", [c / (0.5*(.856e9 + 2*.856e9))])
+@pytest.mark.parametrize("nx", [16])
+@pytest.mark.parametrize("cell_size", [2.0])
+@pytest.mark.parametrize("plot", [False])
+def test_oversampling(exact_u, full_support,
```
Comment (sjperkins) [inline africanus/filters/tests/test_oversampling.py]: @landmanbester This is the small visual test for setting up the filter.

File: africanus/gridding/simple/gridding.py [hunk 3]
```diff
@@ -52,51 +62,50 @@ def numba_grid(vis, uvw, flags, weights, ref_wave,
...
             disc_u = int(np.round(exact_u))
             disc_v = int(np.round(exact_v))
 
-            extent_u = disc_u + half_x
-            extent_v = disc_v + half_y
+            # Filter extents in U and V
+            lower_u = disc_u - half_support
+            lower_v = disc_v - half_support
+            upper_u = disc_u + half_support + 1  # Exclusive
+            upper_v = disc_v + half_support + 1  # Exclusive
```
Comment (sjperkins) [inline africanus/gridding/simple/gridding.py]: @landmanbester I've tried to simplify and make the gridding code even more intuitive by iterating between the lower and upper extents of the filter.

File: africanus/gridding/simple/tests/test_simple_gridder.py (hunk 1 above)
Comment (sjperkins) [reply africanus/gridding/simple/tests/test_simple_gridder.py]: @landmanbester This is probably still the entry point for seeing the discrepancy. Succeeds with oversampling of 1, but fails with 2 and 3.

**Comment by sjperkins** [comment]
@landmanbester I'm tempted to close this PR and remove the branch since the work you've done has superceded this. Should I wait on your new PR or are you happy for me to close this now?

**Comment by landmanbester** [comment]
Just hold on, my local version of gridding_rework has diverged a bit from this one so let's keep it for now. I'll make sure I've taken everything I need from it by the end of today

**Comment by sjperkins** [comment]
Implemented properly in #202 and #204

**Pull request closed without merging**